### PR TITLE
Added LatitudeTest using Gtest

### PR DESCRIPTION
**Pull request by kaitlyndlee**
## Description

Converted the Latitude makefile unit test to gtest. I also added a few comments to Angle and Latitude for clarification on a line of code.

## Related Issue

https://github.com/USGS-Astrogeology/ISIS3/issues/3144

## Motivation and Context

Work contributed to effort to convert all tests to gtest.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

**Comment by kaitlyndlee** [comment]
I used a fixture for the Mapping groups, but I could have used gmock. Do we have a standard for mocking vs. using fixtures?

**Comment by jlaura** [comment]
@kdl222 That is a really good question. @Kelvinrr @kberryUSGS @SgStapleton @scsides @acpaquette @twilson271828 @jessemapel - Thoughts on this? Seems that we should come up with a rule / rule of thumb.

File: isis/src/base/objs/Angle/Angle.cpp [hunk 1]
```diff
@@ -225,7 +225,8 @@ namespace Isis {
       IString msg = "Cannot compare a invalid angles with the < operator";
       throw IException(IException::Programmer, msg, _FILEINFO_);
     }
-
+    // != comparison allows for angles that are considered equal to be treated
+    // as being equal. == operator override uses qFuzzyCompare().
```
Comment (jlaura) [inline isis/src/base/objs/Angle/Angle.cpp:229]: I am confused by this. Is this saying that the code makes != equal == in some cases?

Comment (jlaura) [reply isis/src/base/objs/Angle/Angle.cpp:229]: never mind - I see what this is doing...

**Comment by scsides** [comment]
From what I've seen so far, the fixtures are simpler and easily read. When the amount of data is small and repeated they work great. Large data structures, cubes for example, may fit mocking better.

**Comment by jessemapel** [comment]
I agree with @scsides. If it's a simple object like a PVL object, it should be done as a fixture. If it's a complex object that may be hard to properly initialize, then it should be mocked.

**Comment by SgStapleton** [comment]
Cubes are a clear choice to mock. Would it be worth compiling a list of objects that are better-off mocked? This would not be an exhaustive list, of course, but it would at least serve as a reference for test developers when trying to make those decisions.

**Comment by jessemapel** [comment]
> Cubes are a clear choice to mock. Would it be worth compiling a list of objects that are better-off mocked? This would not be an exhaustive list, of course, but it would at least serve as a reference for test developers when trying to make those decisions.

The googlemock way to do this is to have a central collection of mock objects that can be re-used, maybe `IsisMocks.h`. Anything in that collection should be mocked.

**Review by jessemapel — APPROVED** [review]

**Pull request merged**